go-controller/pkg/ovn/egressip.go

```diff
@@ -268,63 +268,87 @@ func (oc *Controller) syncStaleEgressReroutePolicy(egressIPToPodIPCache map[stri
 }
 
 func (oc *Controller) syncStaleNATRules(egressIPToPodIPCache map[string]sets.String) {
-	natItems, stderr, err := util.RunOVNNbctl(
-		"--format=csv",
-		"--data=bare",
-		"--no-heading",
-		"--columns=_uuid,external_ids,logical_ip",
-		"find",
-		"nat",
-	)
+	nats := []nbdb.NAT{}
+	err := oc.nbClient.WhereCache(func(item *nbdb.NAT) bool {
+		egressIPName, exists := item.ExternalIDs["name"]
+		// Skip nat rows that do not have egressIPName attribute available
+		if !exists {
+			return false
+		}
+		parsedLogicalIP := net.ParseIP(item.LogicalIP).String()
+		podIPCache, exists := egressIPToPodIPCache[egressIPName]
+		return !exists || !podIPCache.Has(parsedLogicalIP)
+	}).List(&nats)
+
 	if err != nil {
-		klog.Errorf("Unable to sync egress IPs, unable to find NAT IDs, stderr: %s, err: %v", stderr, err)
+		klog.Errorf("Unable to sync egress IPs, unable to find NAT IDs, err: %v", err)
 		return
 	}
-	for _, natItem := range strings.Split(natItems, "\n") {
-		if natItem == "" {
-			continue
-		}
-		natFields := strings.Split(natItem, ",")
 
-		UUID := natFields[0]
-		externalID := natFields[1]
-		logicalIP := natFields[2]
+	if len(nats) == 0 {
+		// No stale nat entries to deal with: noop.
+		return
+	}
 
-		parsedLogicalIP := net.ParseIP(logicalIP).String()
-		if !strings.Contains(externalID, "name") {
-			continue
-		}
-		egressIPName := strings.Split(externalID, "=")[1]
+	natUUIDs := make([]string, 0, len(nats))
```

Comment:
There is no need for this slice. You're just using it as "starting point" for your set. You could just build an empty set and add items to it.